### PR TITLE
[commands] add require_var_positional

### PR DESCRIPTION
**Pull request by khazhyk**
If require_var_positonal=True, var_positional arguments require at least
one argument.
Defaults to False, as previous behavior.

### Summary

https://github.com/Rapptz/discord.py/pull/967 for the new branch

### Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/ext/commands/core.py [hunk 1]
```diff
@@ -179,6 +179,10 @@ class Command(_BaseCommand):
         in a completely raw matter. Defaults to ``False``.
     invoked_subcommand: Optional[:class:`Command`]
         The subcommand that was invoked, if any.
+    require_var_positional: :class:`bool`
+        If ``True`` and a varadic positional argument is specified, requires
```
Comment (Rapptz) [inline discord/ext/commands/core.py:183]:
```suggestion
        If ``True`` and a variadic positional argument is specified, requires the
```

File: discord/ext/commands/core.py [hunk 2]
```diff
@@ -179,6 +179,10 @@ class Command(_BaseCommand):
...
+        user specifies at least one argument instead of allowing zero. Defaults
```
Comment (Rapptz) [inline discord/ext/commands/core.py:184]:
```suggestion
        user to specify at least one argument. Defaults to ``False``.
```

File: discord/ext/commands/core.py [hunk 3]
```diff
@@ -179,6 +179,10 @@ class Command(_BaseCommand):
...
+        to ``False``.
```
Comment (Rapptz) [inline discord/ext/commands/core.py:185]: Needs a `.. versionadded:: 1.5` somewhere here.

**Review by Rapptz — CHANGES_REQUESTED** [review]

**Comment by khazhyk** [comment]
Thanks for the review, updated

**Pull request merged**